tests/drivers/uart/uart_basic_api/src/test_uart_config_wide.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023 Jeroen van Dooren
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @addtogroup t_uart_basic
+ * @{
+ * @defgroup t_uart_config test_uart_config_wide
+ * @brief TestPurpose: verify UART configure API settings for wide data support
+ * @details
+ * - Test Steps
+ *   - Configure: test_uart_configure_wide( )
+ *   - Configure Get: test_uart_config_get_wide( )
+ * - Expected Results
+ *   -# When test UART CONFIG Configure, the value of configurations actually
+ *      set will be equal to the original configuration values (from device
+ *      tree or run-time configuration to modify those loaded initially from
+ *      device tree)
+ *   -# When test UART CONFIG Configure Get, the app will get/retrieve the
+ *      value of configurations stored at location and to be passed to UART
+ *      CONFIG Configure
+ * @}
+ */
+
+#include "test_uart.h"
+
+#if DT_NODE_EXISTS(DT_NODELABEL(dut))
+#define UART_NODE DT_NODELABEL(dut)
+#else
+#define UART_NODE DT_CHOSEN(zephyr_console)
+#endif
```

Comment:
If test can work on console, I wonder why we would also implement a dut version

Reply:
Test can't work on console, because twister uses that same console for result reporting? At least twister execution breaks if I reconfigure that console to 9 bit data width.

Reply:
ok, that what I though. Then why the `else` ?

Reply:
Because I didn't want to add all the overlays for all the boards on which this test can be executed. So, the test will build atm, but indeed might not execute. I can't test execution on those boards because I don't have them.
Same construct was already in zephyr/tests/drivers/uart/uart_async_api/src/test_uart.h, so I reused it here.